servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NonPipelinedStreamingHttpConnection.java

```diff
@@ -39,6 +39,7 @@
     @Override
     protected Publisher<Object> writeAndRead(final Publisher<Object> requestStream,
                                              @Nullable final FlushStrategy flushStrategy) {
+        assert connectionContext().protocol().major() >= 2 : "Unexpected protocol version";
```

Comment:
the cancel+close is the only reason why which may not be the case in the future (e.g. for non-pipelined h1 connections we could use this utility, IIRC we don't bcz folks may do `reserveConnection` which doesn't queue requests currently). consider adding comments as to why this restriction exists.